Sources/FioriARKit/ARCards/ViewModels/ARManager.swift

```diff
@@ -33,7 +33,7 @@ public class ARManager {
     
     public init() {
         self.arView = ARView(frame: .zero)
-        self.configureSession(with: ARWorldTrackingConfiguration())
+        do { try self.configureSession(with: ARWorldTrackingConfiguration()) } catch { print(error) }
```

Comment:
is there a reason why we should swallow the error? Otherwise throwable initializer preferred

Reply:
It depends on how far we want the Error to bubble up and if the message of not supporting the simulator is important enough to have the do-try-catch fall on the developer.

The goal was to have some log feedback for the developer to see that FioriARKit does not support the Simulator yet have its execution run for tests. Which the FatalError prevented. If the ARManager doesn't catch it, then it falls on what initializes the ARManager.

The ARManager is currently only used as an internal property of the ARAnnotationViewModel. Either we have the ViewModel swallow the error or have the ViewModel's init throw as well and the catch falls on the developer? Which doesn't feel right and makes the ViewModel's declaration as a StateObject complex. I'm not sure if the developer should know about the try-catch.

Perhaps the design needs reworking or a different way of notifying the developer?

Reply:
Seems we are changing a fatal error to a catchable error but does the app developer actually writes a `do catch` statement? If not then essentially the app developer has to look into the console logs to understand that FioriARKit cannot be used in the simulator during runtime (only tests). Personally, I prefer the fail-fast principle

Reply:
I hear you, just mentioning my thoughts. Yet nothing really "fails". Having the developer implement the try is a fancy way of notifying them that the simulator is not supported yet it still runs. If they run on device it won't fall into the catch and if ran on simulator it will still run and just catch a message for them which they will have to come across anyway.

Perhaps this is something that can be noted heavily in documentation similar to how pointfree notes that [snapshot tests](https://github.com/pointfreeco/swift-snapshot-testing#snapshot-anything) must be on compared using the same OS, gamut, and scale.

If you believe it should be pointed out to the developer at this level I will refactor it.

Reply:
I agree that using a try is ugly

```swift
@StateObject var arModel = try! ARAnnotationViewModel<StringIdentifyingCardItem>()
```

What about the public initializer is not throwable but can be fatal (when using in the simulator) and an internal initializer to allow silent use for tests in the simulator ? Good comprise?

```swift
    public init() {
        self.intialize(canBeFatal: true)
    }

    internal init(canBeFatal: Bool = false) {
        self.intialize(canBeFatal: canBeFatal)
    }

    internal func intialize(canBeFatal: Bool = true) {
        self.arView = ARView(frame: .zero)
        if canBeFatal == true {
            do { try self.configureSession(with: ARWorldTrackingConfiguration()) } catch let error { fatalError(error.localizedDescription) }
        } else {
            do { try self.configureSession(with: ARWorldTrackingConfiguration()) } catch { print(error) }
        }
        self.subscription = self.arView?.scene.subscribe(to: SceneEvents.Update.self) { [unowned self] in
            onSceneUpate?($0)
        }
    }
```

Reply:
This is definitely a good compromise and I adjusted it a bit. The ARAnnotationViewModel also needs to inject the proper ARManager configuration when testing so here is an example of that.  I removed the default value since it can be a bit unclear if its the public init or the fatal init.

If the error throws, but is not fatal the error still prints. But the tests still pass successfully. This could be more descriptive if you'd like.

``` swift
// ARManager
    public init() {
        self.setup(canBeFatal: true)
    }

    internal init(canBeFatal: Bool) {
        self.setup(canBeFatal: canBeFatal)
    }
    
    internal func setup(canBeFatal: Bool = true) {
        
        self.arView = ARView(frame: .zero)

        do {
            try self.configureSession(with: ARWorldTrackingConfiguration())
        } catch {
            if canBeFatal {
                fatalError(error.localizedDescription)
            } else {
                print(error)
            }
        }
        self.subscription = self.arView?.scene.subscribe(to: SceneEvents.Update.self) { [unowned self] in
            onSceneUpate?($0)
        }
    }

// ARAnnotationViewModel
    override public init() {
        super.init()
        self.arManager = ARManager()
        self.arManager.setDelegate(to: self)
        self.arManager.onSceneUpate = self.updateScene(on:)
    }
    
    internal init(arManager: ARManager) {
        super.init()
        self.arManager = arManager
        self.arManager.setDelegate(to: self)
        self.arManager.onSceneUpate = self.updateScene(on:)
    }
```

Reply:
Looks good to me. Did you push the changes @pgobriensap ?